### ISSUE TITLE
Pre-v5.9-client, upgrade 64-bit, change 32-bit, deletes all still-present pre-v5.9-work.

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#255]
**Reported by Rattledagger on 24 Jul 37444549 17:46 UTC**
Old clients don't include the "platform"-tag on applications and results, while v5.10.xx on startup fills-in with the primary-platform-tag if none present.

Meaning, if you're trying-out 64-bit client 1st., all applications/versions will be marked as
`<platform>windows_x86_64</platform>`

If you later switches-back to 32-bit client, any 64-bit results will immediately be deleted with an
```
2007-06-11 16:26:08 [alpha test](BOINC) [App version has unsupported platform windows_x86_64
2007-06-11 16:26:08 [BOINC alpha test](error]) [error] No app version for result: windows_x86_64 515
```
Since many projects including CPDN still haven't upgraded their server to handle 64-bit clients, it's likely many that will install 64-bit, for so switching to 32-bit to report results or continue Trickling... But, this means CPDN-wu's is deleted...

Upgrading to 32-bit client 1st. would not give the same effect, but, any newly-downloaded work marked as 64-bit would also get deleted. Only if project sends work using "alt_platform" can you really safely switch-back from 64-bit to 32-bit.

**Comment by romw** [comment]
**Commented by Rattledagger on 29 Oct 37446983 23:06 UTC**
v5.10.5. The application switched platform as it should, but remember, the result is also marked with a platform, and this still points to the non-existing platform...

  2007-06-12 14:05:14 [alpha test](BOINC) [App version has unsupported platform windows_x86_64; changing to windows_intelx86
  2007-06-12 14:05:14 [BOINC alpha test](error]) [error] No app version for result: windows_x86_64 515

**Comment by romw** [comment]
**Commented by davea on 24 Sep 37447441 08:00 UTC**
(In [12907]) - client: fetch all-projects list at startup if it doesn't exist
- client: if result in statefile is labeled with unsupported platform,
    set its platform to primary
    (this completes the fix of 11 June, hopefully fixes #262)

client/
    client_state.C
    cs_statefile.C

**Comment by romw** [comment]
**Commented by davea on 5 Oct 37447443 01:46 UTC**
 Hopeully fixed by:
     - client: if result in statefile is labeled with unsupported platform,
        set its platform to primary
        (this completes the fix of 11 June, hopefully fixes #262)
    
    client/
        cs_statefile.C
